airbyte-integrations/connectors/source-lexoffice/source_lexware/source.py

```diff
@@ -0,0 +1,247 @@
+
+        As we read all vouchers that are AFTER start_date, we only need a single slice.
+        The read_records method will write the new state containing the newest voucher record
+        """
+        return [{self.cursor_field: start_date.strftime(self.date_format)}]
+
+
+    def stream_slices(
+        self, sync_mode, cursor_field: List[str] = None, stream_state: Mapping[str, Any] = None
+    ) -> Iterable[Optional[Mapping[str, Any]]]:
+        start_date = (
+            datetime.strptime(stream_state[self.cursor_field], self.date_format)
+            if stream_state and self.cursor_field in stream_state
+            else self.start_date
+        )
+        return self._chunk_date_range(start_date)
+
+    #
+    # We only return a voucher without any meta data
+    #
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        """
+        We return the first (and only since page size is 1) voucher here or an empty object if none
+        """
+        resultList = response.json().get("content", [])
+        result = resultList[0] if len(resultList) > 0 else {}
+        yield result
+
+
+# Source
+class SourceLexware(AbstractSource):
+    def check_connection(self, logger, config) -> Tuple[bool, any]:
+        """
+        A connection check to validate that the user-provided config can be used to connect to the underlying API
+
+        :param config:  the user-input config object conforming to the connector's spec.yaml
+        :param logger:  logger object
+        :return Tuple[bool, any]: (True, None) if the input config can be used to connect to the API successfully, (False, error) otherwise.
+        """
+
+        apikey = config["apikey"]
+        r = requests.get("https://api.lexoffice.io/v1/ping", headers={"Authorization": "Bearer " + apikey})
+
+        # check for status_code
+        if r.status_code == 200:
+            return True, None
+        else:
+            return False, "Could not verify connection to lexoffice API. Did you provide the correct API key?"
+
+    def streams(self, config: Mapping[str, Any]) -> List[Stream]:
+        """
+        :param config: A Mapping of the user input configuration as defined in the connector spec.
+        """
+        auth = TokenAuthenticator(config["apikey"])  # Oauth2Authenticator is also available if you need oauth support
+
+        # Parse the date from a string into a datetime object
+        date_format = "%Y-%m-%d"  # %Y-%m-%dT%H%M%S.%f%z = 2022-09-11T04: 21: 36.000+02: 00
+        date = config.get("start_date", None)
+        start_date = datetime.strptime(date, date_format) if date is not None and date != "" else None
+
+        voucherType = config.get("voucher_type", "any").replace(" ", "")
+        voucherStatus = config.get("voucher_status", "any").replace(" ", "")
+
+        return [VoucherList(authenticator=auth, voucherType=voucherType, voucherStatus=voucherStatus, start_date=start_date)]
```

Comment:
Are you going to add others streams https://developers.lexoffice.io/docs/#invoices-endpoint ?

Reply:
This is not planned in near future, since the voucher already contains all relevent documents that are required to synchronize accounting activities. Everything else is meta data or belongs do another domain (e.g. HR). However, in future, additional streams might be added.